network-api/networkapi/wagtailpages/forms.py

```diff
@@ -24,3 +24,16 @@ def clean_parent(self):
             if self.instance == parent:
                 self.add_error("parent", "A category cannot be a parent of itself.")
         return parent
+
+
+# Max number validation for blog page topics. Since this is a parental m2m relationship,
```

Comment:
Hi @tbrlpld here is a link to the SA article I was referencing earlier, https://stackoverflow.com/a/37865151, though I am not sure if it is worth adding this link inside the codes comments. So I updated the comment to try and explain why we are doing this as best as possible

Reply:
@danielfmiranda The SO thread you are linking is about the default Django `ManyToManyField` and revolves around the `ValueError` with message "instance needs to have a primary key value before a many-to-many relationship can be used.". 

The topics are not using the default Django `ManyToManyField`. They are using the `modelcluster.fields.ParentalManyToManyField`. The point of `django-modelcluster` is exactly to address the above mentioned `ValueError` that requires all models to be saved before a many-to-many relation can be established. 

This is not the error we have been seeing though. When you demoed it to me (and I have done some tests myself and been able to reproduce the issue), we have seen the `ValidationError` that we raised in the `BlogPage.clean()` method was not caught and turned into a form error. Instead, our `ValidationError` led to a 500 error, which is unexpected. 

I have raised that with the Wagtail team and am waiting for a response. 

Since the comment and SO thread are about a different kind of error and not the reason we are going with the form, it might be best to remove the comment to avoid confusion in the future.